craft_providers/lxd/lxc.py

```diff
@@ -245,6 +245,51 @@ def config_set(
                 details=errors.details_from_called_process_error(error),
             ) from error
 
+    def copy(
+        self,
+        *,
+        source_remote: str = "local",
+        source_instance_name: str,
+        destination_remote: str = "local",
+        destination_instance_name: str,
+        project: str = "default",
```

Comment:
Oh here's another argument pro-`Optional[str]`: the "current" project might be something other that `default`, right? I think you can change the target project of commands with `lxc project switch`

Reply:
The public methods (`delete`, `file_push`, `list`, etc) all have `project: str = "default"`.
The private method is the only one to have `project: Optional[str] = None`.
So from a consistency standpoint, I think it is correct as-is.   

I think your second argument is still true - all the public methods could make this parameter optional. This would belong in a separate PR.